aws/rust-runtime/aws-config/src/profile/parser.rs

```diff
@@ -235,21 +240,29 @@ mod test {
 
     #[test]
     fn empty_source_empty_profile() {
-        let source = Source {
-            config_file: File {
-                path: "~/.aws/config".to_string(),
-                contents: "".into(),
-            },
-            credentials_file: File {
-                path: "~/.aws/credentials".to_string(),
-                contents: "".into(),
-            },
-            profile: "default".into(),
-        };
+        let source = make_source(ParserInput {
+            config_file: Some("".to_string()),
+            credentials_file: Some("".to_string())
+        });
+
         let profile_set = ProfileSet::parse(source).expect("empty profiles are valid");
         assert!(profile_set.is_empty());
     }
 
+    #[test]
+    fn profile_names_are_exposed() {
+        let source = make_source(ParserInput {
+            config_file: Some("[profile foo]\n[profile bar]".to_string()),
+            credentials_file: Some("".to_string())
+        });
+
+        let profile_set = ProfileSet::parse(source).expect("profiles loaded");
+
+        let mut profile_names: Vec<&str> = profile_set.profiles().collect();
```

Comment:
```suggestion
        let mut profile_names: Vec<_> = profile_set.profiles().collect();
```

You only need to specify the container when collecting (Vec is container in this case)

Reply:
OK cool - will fix.